src/metalookup/features/malicious_extensions.py

```diff
@@ -0,0 +1,226 @@
+            ".lok",
+            ".hta",
+            ".vba",
+            ".xltm",
+            ".atm",
+            ".xtbl",
+            ".txs",
+            ".xlsm",
+            ".mjz",
+            ".mfu",
+            ".wsf",
+            ".cih",
+            ".xnt",
+            ".capxml",
+            ".sfx",
+            ".fjl",
+            ".cmd",
+            ".msh",
+            ".aut",
+            ".ws",
+            ".tti",
+            ".dlb",
+            ".msh1",
+            ".ozd",
+            ".fuj",
+            ".exe",
+            ".class",
+            ".386",
+            ".qit",
+            ".ps2",
+            ".delf",
+            ".cla",
+            ".ps1xml",
+            ".bkd",
+            ".bin",
+            ".dev",
+            ".cc",
+            ".sys",
+            ".dx",
+            ".vbx",
+            ".bup",
+            ".vxd",
+            ".rsc_tmp",
+            ".spam",
+            ".tps",
+            ".htm",
+            ".wsh",
+            ".bll",
+            ".sop",
+            ".wsc",
+            ".bxz",
+            ".jar",
+            ".tsa",
+            ".msi",
+            ".pcx",
+            ".vbe",
+            ".smm",
+            ".rhk",
+            ".dli",
+            ".application",
+            ".let",
+            ".pid",
+            ".upa",
+            ".msh1xml",
+            ".ce0",
+            ".psc2",
+            ".msh2",
+            ".lpaq5",
+            ".ctbl",
+            ".boo",
+            ".buk",
+            ".hts",
+            ".sldm",
+            ".bat",
+            ".smtmp",
+            ".dllx",
+            ".ppsm",
+            ".docm",
+            ".bhx",
+            ".scf",
+            ".fnr",
+            ".pptm",
+            ".drv",
+            ".doc",
+            ".vzr",
+            ".ssy",
+            ".scr",
+            ".dotm",
+            ".s7p",
+            ".ceo",
+            ".tmp",
+            ".lik",
+            ".lnk",
+            ".pgm",
+            ".dll",
+            ".oar",
+            ".bqf",
+            ".zvz",
+            ".dbd",
+            ".vexe",
+            ".potm",
+            ".\u202e",
+        }
+
+    async def setup(self):
+        pass
+
+    async def extract(self, site: WebsiteData, executor: Executor) -> tuple[StarCase, Explanation, set[str]]:
+        def extract_extensions(raw_links: list[str]) -> set[str]:
+            def suffix(link: str):
+                return os.path.splitext(urlparse(link)[2])[-1]
+
+            return {suffix(link) for link in raw_links} - {""}
+
+        extensions = extract_extensions(raw_links=site.raw_links)
+        malicious = {extension for extension in extensions if extension in self.malicious_extensions}
+
+        if len(malicious) > 0:
+            return StarCase.ZERO, FOUND_POTENTIALLY_MALICIOUS_EXTENSION, malicious
+
+        harmless = {extension for extension in extensions if extension in self.dangerous_extensions}
```

Comment:
So an extension is harmless if it is in dangerous_extensions?!?! That is a surprise :D